lib/improver/tests/nowcasting/lightning/test_NowcastLightning.py

```diff
@@ -0,0 +1,894 @@
+                                            None)
+        self.assertArrayAlmostEqual(result.data, expected.data)
+
+    def test_vii_large(self):
+        """Test that ApplyIce is being called"""
+        # Set lightning data to zero so it has a Null impact
+        self.vii_cube.data[:, 7, 7] = 1.
+        self.ltng_cube.data[0, 7, 7] = -1.
+        self.fg_cube.data[0, 7, 7] = 0.
+        # No halo - we're only testing this method.
+        plugin = Plugin(0.)
+        expected = self.fg_cube.copy()
+        # expected.data contains all ones except:
+        expected.data[0, 7, 7] = 0.9
+        result = plugin._modify_first_guess(self.cube,
+                                            self.fg_cube,
+                                            self.ltng_cube,
+                                            self.precip_cube,
+                                            self.vii_cube)
+        self.assertArrayAlmostEqual(result.data, expected.data)
+
+    def test_null(self):
+        """Test that large precip probs and -1 lrates have no impact"""
+        # Set prob(precip) data for lowest threshold to to 0.1, the highest
+        # value that has no impact.
+        self.precip_cube.data[0, 0, 7, 7] = 0.1
+        # Set lightning data to -1 so it has a Null impact
+        self.ltng_cube.data = np.full_like(self.ltng_cube.data, -1.)
+        # No halo - we're only testing this method.
+        plugin = Plugin(0.)
+        expected = self.fg_cube.copy()
+        # expected.data should be an unchanged copy of fg_cube.
+        result = plugin._modify_first_guess(self.cube,
+                                            self.fg_cube,
+                                            self.ltng_cube,
+                                            self.precip_cube,
+                                            None)
+        self.assertArrayAlmostEqual(result.data, expected.data)
+
+    def test_lrate_large(self):
+        """Test that large lightning rates increase lightning risk"""
+        expected = self.fg_cube.copy()
+        # expected.data contains all ones
+        # Set precip data to 1. so it has a Null impact
+        # Set prob(precip) data for lowest threshold to to 1., so it has a Null
+        # impact when lightning is present.
+        self.precip_cube.data[0, 0, 7, 7] = 1.
+        # Set first-guess data zero point that will be increased
+        self.fg_cube.data[0, 7, 7] = 0.
+        # No halo - we're only testing this method.
+        plugin = Plugin(0.)
+        result = plugin._modify_first_guess(self.cube,
+                                            self.fg_cube,
+                                            self.ltng_cube,
+                                            self.precip_cube,
+                                            None)
+        self.assertArrayAlmostEqual(result.data, expected.data)
+
+    def test_lrate_halo(self):
+        """Test that lightning in 50 km vicinity / halo (encoded as lightning
```

Comment:
I don't know if all these docstrings are up to date, or what the halo means. Can you update them?

Reply:
Done